print package details at end

esc opened the pull request:
One of the most common questions when running integration tests is:
"what were the final package versions that you ran with?". Since this
is so common, we now print the final versions of the relevant packages
at the end of the test-run.

esc committed:
print package details at end

One of the most common questions when running integration tests is:
"what were the final package versions that you ran with?". Since this
is so common, we now print the final versions of the relevant packages
at the end of the test-run.

stuartarchibald reviewed (dismissed):
Thanks for the patch.

esc committed:
change lookup

Instead of using a pipe to grep, we use the regex parameter for conda
list to show a subset of the packages.

esc committed:
switch to searching on the captured output

It will look like:

```
::>> running: 'conda list -n umap '
llvmlite                  0.35.0rc1                py38_0    numba/label/dev
numba                     0.52.0rc1       np1.11py3.8h6440ff4_g3a682bd_0    numba/label/dev
umap-learn                0.4.5                     dev_0    <develop>
```

esc commented:
@stuartarchibald the implementation now searches the captured output of the command with Python.

stuartarchibald commented on texasbbq.py:
```diff
@@ -437,6 +437,16 @@ def print_environment_details(target):
 
+def print_package_details(source_name, target_name):
+    lines= execute("conda list -n {}".format(target_name),
+                   capture=True).decode("utf-8").split("\n")
```
```suggestion
    lines = execute("conda list -n {}".format(target_name),
                    capture=True).decode("utf-8").split("\n")
```

esc replied:
Yeah, texasbbq needs to be developed outside of a conda-env to ensure an existing, activated env doesn't clobber the envs activated by the tool itself. I didn't check to see if flake8 was installed so I let this one slip.

stuartarchibald reviewed (commented):
Thanks for fixing this to not use the shell, style typo else looks good.

esc committed:
Update texasbbq.py

Co-authored-by: stuartarchibald <stuartarchibald@users.noreply.github.com>
